modules/orientdb/src/main/java/org/testcontainers/containers/OrientDBContainer.java

```diff
@@ -0,0 +1,181 @@
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Optional;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+/**
+ * @author robfrank
+ */
+public class OrientDBContainer extends GenericContainer<OrientDBContainer> {
+
+    private static final String DEFAULT_IMAGE_NAME = "orientdb";
+
+    private static final String DEFAULT_TAG = "3.0.24-tp3";
+
+    private static final String DEFAULT_USERNAME = "admin";
+
+    private static final String DEFAULT_PASSWORD = "admin";
+
+    private static final String DEFAULT_DATABASE_NAME = "testcontainers";
+
+    private static final int DEFAULT_BINARY_PORT = 2424;
+
+    private static final int DEFAULT_HTTP_PORT = 2480;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrientDBContainer.class);
+
+    private static final String DOCKER_IMAGE_NAME = DEFAULT_IMAGE_NAME + ":" + DEFAULT_TAG;
+
+    private String databaseName;
+
+    private String username;
+
+    private String password;
+
+    private OrientDB orientDB;
+
+    private ODatabaseSession session;
+
+    private Optional<String> scriptPath = Optional.empty();
+
+
+    public OrientDBContainer() {
+        this(DOCKER_IMAGE_NAME);
+    }
+
+    public OrientDBContainer(@NonNull String dockerImageName) {
+        super(dockerImageName);
+
+        username = DEFAULT_USERNAME;
+        password = DEFAULT_PASSWORD;
+        databaseName = DEFAULT_DATABASE_NAME;
+
+        WaitStrategy waitForHttp = new HttpWaitStrategy()
+            .forPort(DEFAULT_HTTP_PORT)
+            .forStatusCodeMatching(response -> response == HTTP_OK);
+
+        waitStrategy = new WaitAllStrategy()
+            .withStrategy(Wait.forListeningPort())
+            .withStrategy(waitForHttp)
+            .withStartupTimeout(Duration.ofMinutes(2));
+    }
+
+    @Override
+    protected void configure() {
+
+        addExposedPorts(DEFAULT_BINARY_PORT, DEFAULT_HTTP_PORT);
+        addEnv("ORIENTDB_ROOT_PASSWORD", "root");
+    }
+
+
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getTestQueryString() {
+        return "SELECT FROM V";
+    }
+
+    public OrientDBContainer withDatabaseName(final String databaseName) {
+        this.databaseName = databaseName;
+        return self();
+    }
+
+    public OrientDBContainer withUsername(final String username) {
+        this.username = username;
+        return self();
+    }
+
+    public OrientDBContainer withPassword(final String password) {
+        this.password = password;
+        return self();
+    }
```

Comment:
This username and password don't seem to be used to configure the container. In the JDBC database classes, username/password are passed through to the container via environment variables, to tell the container's bootstrap script to create that account.

It [doesn't look like](https://hub.docker.com/_/orientdb) OrientDB supports this natively. Line 147 _might_ be an opportunity to create the database and give access to the provided `username`, but as far as I can tell it's just expecting those credentials to work on line 150.

Is that accurate?

If these methods don't ensure that the account exists, I'd prefer to remove them to avoid confusion, TBH.

Reply:
You're right. I misunderstood the purpose of the member variables of the Container class, mixing server and client variables.  I moved to the getSession method.